src/App.tsx

```diff
@@ -15,36 +18,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const IS_SORTED_ALPHABETICALLY = 'alphabetically';
+const IS_SORTED_BY_LENGTH = 'length';
+
+function getPreparedGoods(
+  goods: string[],
+  { sortType, isReversed }: { sortType: string; isReversed: boolean },
+) {
+  let preparedGoods = [...goods];
+
+  switch (sortType) {
+    case IS_SORTED_ALPHABETICALLY:
+      preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case IS_SORTED_BY_LENGTH:
+      preparedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortType,
+    isReversed,
+  });
+
+  const toggleReverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const reset = () => {
+    setSortType('');
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortType !== IS_SORTED_ALPHABETICALLY,
+          })}
+          onClick={() => setSortType(IS_SORTED_ALPHABETICALLY)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortType !== IS_SORTED_BY_LENGTH,
+          })}
+          onClick={() => setSortType(IS_SORTED_BY_LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={toggleReverse}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {sortType || isReversed ? (
+          <button type="button" className="button is-danger" onClick={reset}>
+            Reset
+          </button>
+        ) : null}
```

Comment:
Better to create a variable for sortType || isReversed
And instead of ternary operation better to use here like isResetButtonVisible & (component)

Reply:
Good point, fix this